src/phantom/_base.py

```diff
@@ -125,13 +131,26 @@ def __init_subclass__(
         predicate: Predicate[T] | None = None,
         bound: type[T] | None = None,
         abstract: bool = False,
+        use_docstring: Optional[bool] = None,
         **kwargs: Any,
     ) -> None:
+        if kwargs:
+            raise RuntimeError("Unknown phantom type argument(s): {kwargs}")
```

Comment:
issue: I don't think we should add this check, any arguments falling through will be handled by the super call.

Reply:
Well I personally added it after wasting 10 minutes trying to understand why my keyword argument was not taken and it turned out that I had a typo - because the default error by the super call is absolutely not helpful and says (.... takes no keword arguments...) needlessly confusing me.

As the immediate super class does not have an `init_subclass` method, I decided to put it where it is to prevent passing it to the default implementaion.

I personally believe that such a message improves ergonomy over generic messages by just letting some method upstream choke on it. If you strongly disagree, I can remove it.